bridge_adaptivity/module/views.py

```diff
@@ -147,28 +170,53 @@ class GroupCreate(BaseGroupView, SetUserInFormMixin, GroupEditFormMixin, CreateV
 
 
 @method_decorator(login_required, name='dispatch')
-class GroupDetail(BaseGroupView, DetailView):
+class GroupDetail(LinkObjectsMixin, BaseGroupView, DetailView):
     context_object_name = 'group'
+    link_form_class = AddCollectionGroupForm
+    link_object_name = 'collection'
+
+    def get_link_form_kwargs(self):
+        return dict(user=self.request.user, group=self.object)
+
+    def get_link_action_url(self):
+        return reverse('module:collection-add')
+
+    def get_has_available_objects(self, form):
+        return form.fields['collections'].queryset.exists()
 
     def get_context_data(self, **kwargs):
         context = super(GroupDetail, self).get_context_data(**kwargs)
-        context.update({'bridge_host': settings.BRIDGE_HOST})
+        context.update({
+            'bridge_host': settings.BRIDGE_HOST,
+        })
         return context
 
 
+class AddCollectionInGroup(JsonResponseMixin, FormView):
+    template_name = 'module/modals/course_add_group.html'
+    model = CollectionGroup.collections.through
```

Comment:
Why as a model we need to use an intermediate table instead of `CollectionGroup.collections` or even `CollectionGroup`?

Reply:
because this view operates with only this model. but maybe model param is not needed here... ? I will check it